src/toxcore/dht/server/mod.rs

```diff
@@ -2356,36 +2406,49 @@ mod tests {
         }
     }
 
+    // handle_lan_discovery
     #[test]
-    fn server_handle_lan_discovery_test() {
+    fn handle_lan_discovery() {
         let (alice, _precomp, bob_pk, bob_sk, rx, addr) = create_node();
 
         let lan = DhtPacket::LanDiscovery(LanDiscovery { pk: bob_pk });
 
-        assert!(alice.handle_packet(lan, addr).wait().is_ok());
+        alice.handle_packet(lan, addr).wait().unwrap();
 
         let (received, _rx) = rx.into_future().wait().unwrap();
         let (packet, addr_to_send) = received.unwrap();
 
         assert_eq!(addr_to_send, addr);
 
         let nodes_req = unpack!(packet, DhtPacket::NodesRequest);
-        let _nodes_req_payload = nodes_req.get_payload(&bob_sk).unwrap();
+        let nodes_req_payload = nodes_req.get_payload(&bob_sk).unwrap();
 
-        assert_eq!(nodes_req.pk, alice.pk);
+        assert_eq!(nodes_req_payload.pk, alice.pk);
     }
 
     #[test]
-    fn server_handle_lan_discovery_for_ourselves_test() {
+    fn handle_lan_discovery_for_ourselves() {
         let (alice, _precomp, _bob_pk, _bob_sk, _rx, addr) = create_node();
 
         let lan = DhtPacket::LanDiscovery(LanDiscovery { pk: alice.pk });
 
-        assert!(alice.handle_packet(lan, addr).wait().is_ok());
+        alice.handle_packet(lan, addr).wait().unwrap();
     }
 
     #[test]
-    fn refresh_onion_key_test() {
+    fn handle_lan_discovery_when_disabled() {
+        let (mut alice, _precomp, _bob_pk, _bob_sk, _rx, addr) = create_node();
+
+        alice.enable_lan_discovery(false);
+        assert_eq!(alice.lan_discovery_enabled, false);
+
+        let lan = DhtPacket::LanDiscovery(LanDiscovery { pk: alice.pk });
+
+        alice.handle_packet(lan, addr).wait().unwrap();
```

Comment:
As I said before, you should also check that no actions are done when lan is disabled

Reply:
rx is empty